test/e2e/registration_service_test.go

```diff
@@ -39,15 +45,393 @@ func (s *registrationServiceTestSuite) TestLandingPageReachable() {
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+
+		// Retrieve unauthorized http status code.
+		assert.Equal(s.T(), http.StatusUnauthorized, resp.StatusCode)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		// Check token error.
+		tokenErr := mp["error"]
+		require.IsType(s.T(), tokenErr, "")
+		require.Equal(s.T(), "token contains an invalid number of segments", tokenErr.(string))
+	})
+	s.Run("get signup exp token 401 Unauthorized", func() {
+		expClaim1 := authsupport.WithExpClaim(time.Now().Add(-60 * time.Second))
+
+		// Not identical to the token used in POST signup - should return resource not found.
+		token1, err := authsupport.GenerateSignedE2ETestToken(*identity0, emailClaim0, iatClaim0, expClaim1)
+		require.NoError(s.T(), err)
+
+		// Call signup endpoint with an valid token.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/signup", nil)
+		require.NoError(s.T(), err)
+		req.Header.Set("Authorization", "Bearer "+token1)
+		req.Header.Set("content-type", "application/json")
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		// Check token error.
+		tokenErr := mp["error"]
+		require.IsType(s.T(), tokenErr, "")
+		require.Contains(s.T(), tokenErr.(string), "token is expired by ")
+
+		assert.Equal(s.T(), http.StatusUnauthorized, resp.StatusCode)
+	})
+	s.Run("get signup 200 OK not approved", func() {
+		// Call signup endpoint with an valid token.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/signup", nil)
+		require.NoError(s.T(), err)
+		req.Header.Set("Authorization", "Bearer "+token0)
+		req.Header.Set("content-type", "application/json")
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		mpStatus, ok := mp["status"].(map[string]interface{})
+		assert.True(s.T(), ok)
+
+		assert.Nil(s.T(), mp["CompliantUsername"])
+		assert.Equal(s.T(), identity0.Username, mp["username"])
+		require.IsType(s.T(), false, mpStatus["ready"])
+		assert.False(s.T(), mpStatus["ready"].(bool))
+		assert.Equal(s.T(), "PendingApproval", mpStatus["reason"])
+	})
+	s.Run("get signup 200 OK approved", func() {
+		userSignup, err := s.awaitility.Host().WaitForUserSignup(identity0.ID.String(), wait.UntilUserSignupHasConditions(pendingApproval()...))
```

Comment:
You are relying on another subtest here. If we delete "post signup test accepted" subtest then this one will start failing and it's not obvious.

Let's merge all relevant tests which deals with the same successful usersignup.

The scenario I mentioned above:

> - Post a signup.
> - Verify that the corresponding UserSignup was created in the cluster with proper status conditions.
> - Check that GET /sigup returns the signup with expected status.
> - Approve the usersignup.
> - Wait for the MUR provisioning.
> - Checking that now GET /sigup returns the status with Provisioned.

is not a list of individual tests. It's one e2e test. Let's create a new test function, something like `TestSignupOK()` and tests all the steps from above in this test (not individually! but as one e2e flow). And move all current failure tests from the current `TestSignup()` to `TestSignupFails()` or something like this.